tendrl/alerting/etcd/manager.py

```diff
@@ -0,0 +1,137 @@
+import ast
+import etcd
+import json
+import logging
+from tendrl.alerting.definitions.alerting \
+    import data as def_data
+from tendrl.alerting.etcd.exceptions import EtcdError
+from tendrl.alerting.persistence.tendrl_definitions \
+    import TendrlDefinitions
+from tendrl.common.config import ConfigNotFound
+from tendrl.common.config import TendrlConfig
+from tendrl.common.etcdobj.etcdobj import Server as etcd_server
+from tendrl.common.singleton import to_singleton
+import time
+import yaml
+
+config = TendrlConfig()
+LOG = logging.getLogger(__name__)
+
+
+@to_singleton
+class EtcdManager(object):
+    def __init__(self):
+        try:
+            etcd_kwargs = {
+                'port': int(config.get("common", "etcd_port")),
+                'host': config.get("common", "etcd_connection")
+            }
+            self.etcd_server = etcd_server(etcd_kwargs=etcd_kwargs)
+        except (
+            ConfigNotFound,
+            etcd.EtcdKeyNotFound,
+            etcd.EtcdConnectionFailed,
+            ValueError,
+            SyntaxError,
+            etcd.EtcdException,
+            TypeError
+        ) as ex:
+            raise EtcdError(str(ex))
+
+    def write_configs(self):
+        try:
+            confs = {
+                'api_server_addr': config.get(
+                    "alerting",
+                    "api_server_addr"
+                ),
+                'api_server_port': config.get(
+                    "alerting",
+                    "api_server_port"
+                )
+            }
+            self.etcd_server.client.write(
+                '/_tendrl/config/alerting',
+                confs
+            )
+        except ConfigNotFound as ex:
+            raise EtcdError(str(ex))
+
+    def get_alerts(self, filters=None):
+        alerts_arr = []
+        try:
+            alerts = self.etcd_server.client.read('/alerts', recursive=True)
+        except etcd.EtcdKeyNotFound:
+            return alerts_arr
+        except (
+            etcd.EtcdConnectionFailed,
+            ValueError,
+            SyntaxError,
+            TypeError
+        ) as ex:
+            LOG.error(
+                'Failed to fetch alerts. Error %s' % str(ex),
+                exc_info=True
+            )
+            raise EtcdError(str(ex))
+        for child in alerts._children:
+            alerts_arr.append(json.loads(child['value']))
```

Comment:
why not filter in this for loop itself? you are basically iterating through the same data set twice (L82)